### ISSUE TITLE
ImportErrors are not thrown for missing objects

### ISSUE THREAD

**Issue by lmmentel** [lmmentel/mendeleev#149]
## Describe the bug

Spotted by @kalvdans in this [comment](https://github.com/lmmentel/mendeleev/pull/148#issuecomment-2108593358).

A recent fix #148 introduced the mechanism by which when importing arbitrary names from `mendeleev` won't throw an `ImportError`

## To Reproduce

Run
```python
# whatever does not exist in the package
from mendeleev import whatever 
```

## Expected behavior
Import Errors are thrown for non-existing objects

## Specification

- mendleev version 0.16.1
- python version 3.12
- platform all

**Comment by lmmentel** [comment]
@kalvdans do you have any suggestions on how to tackle this?

My initial thought is to delegate import responsibilities back to `getattr` somewhere at the end of overridden `__getattr__` function in the `mendeleev.__init__.py`. Not entirely sure how at this point.

**Comment by kalvdans** [comment]
> @kalvdans do you have any suggestions on how to tackle this?

I already gave my advice in https://github.com/lmmentel/mendeleev/pull/121#issuecomment-1612460655 to not mix static and dynamic content in the same namespace. My only advice if you still insist, is to write some unit tests for the behaviour you want, like
```python
def test_nonexisting():
    with pytest.raises(AttributeError):
        _ = mendeleev.Unobtanium
```

**Comment by lmmentel** [comment]
Thanks! Have you thought about an alternative for the  current way it's implemented? I would very much like to keep the shorthand import of elements by symbol without taking too much of a performance hit. BTW this has been available in mendeleev since [v0.4.0](https://mendeleev.readthedocs.io/en/stable/changes_link.html#v0-4-0-22-11-2017).

Good point on the tests, I've added a few to reduce the risk of tripping over similar issues in the future.
